MachRegister/base_registers/x86_64.cpp

```diff
@@ -0,0 +1,50 @@
+#include "basereg_check.h"
+#include "registers/x86_64_regs.h"
+
+int main() {
+  BASEREG_CHECK(x86_64::ax, x86_64::rax);
+  BASEREG_CHECK(x86_64::eax, x86_64::rax);
+  BASEREG_CHECK(x86_64::rax, x86_64::rax);
+
+  BASEREG_CHECK(x86_64::sp, x86_64::rsp);
+  BASEREG_CHECK(x86_64::esp, x86_64::rsp);
+  BASEREG_CHECK(x86_64::rsp, x86_64::rsp);
```

Comment:
check byte-sized register's base; the instruction pointer, the middle (rdi) highest (r15) GPRs for numbering problems:

```suggestion
  BASEREG_CHECK(x86_64::al, x86_64::rax);
  BASEREG_CHECK(x86_64::ah, x86_64::rax);
  BASEREG_CHECK(x86_64::ax, x86_64::rax);
  BASEREG_CHECK(x86_64::eax, x86_64::rax);
  BASEREG_CHECK(x86_64::rax, x86_64::rax);

  BASEREG_CHECK(x86_64::sp, x86_64::rsp);
  BASEREG_CHECK(x86_64::esp, x86_64::rsp);
  BASEREG_CHECK(x86_64::rsp, x86_64::rsp);
  
  BASEREG_CHECK(x86_64::dil, x86_64::rdi);
  BASEREG_CHECK(x86_64::di, x86_64::rdi);
  BASEREG_CHECK(x86_64::edi, x86_64::rdi);
  BASEREG_CHECK(x86_64::rdi, x86_64::rdi);
  
  BASEREG_CHECK(x86_64::r15b, x86_64::r15);
  BASEREG_CHECK(x86_64::r15w, x86_64::r15);
  BASEREG_CHECK(x86_64::r15d, x86_64::r15);
  BASEREG_CHECK(x86_64::r15, x86_64::r15);

  BASEREG_CHECK(x86_64::eip, x86_64::rip);
  BASEREG_CHECK(x86_64::rip, x86_64::rip);
```

Reply:
I added lots more. It turned out `x86_64::eip -> rip` was broken.